### PR TITLE
VA-4212 Publish to Social Jobs

### PR DESCRIPTION
**Pull request by howardrigberg**
#### Issue
- https://vimean.atlassian.net/browse/VA-4212
- https://vimean.atlassian.net/browse/VA-4213

#### Summary
Added support for GET and PUT requests to /videos/{video_id}/publish_to_social

#### How to Test
Test setup is a bit complicated. See me IRL if you want instructions for testing in the app.

### CODE REVIEW

File: models/src/main/java/com/vimeo/networking2/PublishToFacebookPost.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Represents the required data for a Facebook post.
+ */
+@JsonClass(generateAdapter = true)
+data class PublishToFacebookPost(
+
+        /**
+         * The title of the post as it will appear on Facebook.
+         */
+        @Json(name = "title")
+        val title: String,
```
Comment (anthonycr) [inline models/src/main/java/com/vimeo/networking2/PublishToFacebookPost.kt]: This and all properties in this class should be nullable with a default value of null. This comment also applies to
- `PublishToLinkedInPost`
- `PublishToTwitterPost`
- `PublishToYouTubePost`

Unless the API ignores field filters for these non null properties they should be made null.

File: models/src/main/java/com/vimeo/networking2/PublishToYouTubePost.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        @Json(name = "tags")
+        val tags: List<String>? = null,
+
+        /**
+         * The privacy option for this video on YouTube.
+         * public: The video will be publicly available.
+         * private: The video will not be publicly available, and only viewable to the owner.
+         */
+        @Json(name = "privacy")
+        val privacy: String,
```
Comment (anthonycr) [inline models/src/main/java/com/vimeo/networking2/PublishToYouTubePost.kt]: Should have a corresponding enum if I understand the documentation correclty

**Review by anthonycr — CHANGES_REQUESTED** [review]

**Review by WhosNickDoglio — APPROVED** [review]
The only change I was going to mention was the enum for Privacy that @anthonycr already mentioned. 

Everything else looks good. 👍

**Comment by anthonycr** [comment]
@howardrigberg I can review after you resolve the merge conflicts.

**Comment by howardrigberg** [comment]
@anthonycr  Thanks for the review. I moved the param objects into a new package called "params" to distinguish them from the other models. I added the necessary enum and used  `@SerializedName` to ensure that all params will get serialized to the body of the put request correctly.

@WhosNickDoglio I renamed your `PublishJobDestinations` class to `PublishJobAttempts` to resolve a merge conflict and give the class a name that more accurately reflects its purpose. Please let me know if it passes your functional tests and that you agree with my changes.

**Comment by WhosNickDoglio** [comment]
> @WhosNickDoglio I renamed your `PublishJobDestinations` class to `PublishJobAttempts` to resolve a merge conflict and give the class a name that more accurately reflects its purpose. Please let me know if it passes your functional tests and that you agree with my changes.

@howardrigberg  No issues with name changes and just tested and everything looks good. 👍

File: models/build.gradle [hunk 3]
```diff
@@ -19,6 +19,8 @@ dependencies {
     kapt "com.squareup.moshi:moshi-kotlin-codegen:1.8.0"
     implementation "com.squareup.moshi:moshi:1.8.0"
 
+    api "com.squareup.retrofit2:converter-gson:2.5.0"
```
Comment (anthonycr) [inline models/build.gradle]: Please use Moshi, we are not using Gson going forward.

**Review by anthonycr — CHANGES_REQUESTED** [review]

**Review by anthonycr — APPROVED** [review]
LGTM!

**Pull request merged**